<?xml version="1.0"?>
<launch>
  <!-- arguments -->
  <arg name="nodelet_manager" default="rs_m1_nodelet_manager" />
  <arg name="points_topic" default="/os1_points" />
  <arg name="imu_topic" default="/imu/data_raw" />

  <!-- in case you use velodyne_driver, comment out the following line -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/>

  <!-- globalmap_server_nodelet -->
  <node pkg="nodelet" type="nodelet" name="globalmap_server_nodelet" args="load hdl_localization/GlobalmapServerNodelet $(arg nodelet_manager)">
    <rosparam command="load" file="$(find hdl_localization)/config/hdl_loc_rs_m1.yaml" />
  </node>

  <!-- hdl_localization_nodelet -->
  <node pkg="nodelet" type="nodelet" name="hdl_localization_nodelet" args="load hdl_localization/HdlLocalizationNodelet $(arg nodelet_manager)">
    <remap from="/velodyne_points" to="$(arg points_topic)" />
    <remap from="/gpsimu_driver/imu_data" to="$(arg imu_topic)" />
    <rosparam command="load" file="$(find hdl_localization)/config/hdl_loc_rs_m1.yaml" />
  </node>

  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find hdl_localization)/rviz/hdl_localization.rviz" />
</launch>
